src/main/java/teammates/storage/search/SearchManager.java

```diff
@@ -69,21 +72,32 @@ SolrQuery getBasicQuery(String queryString) {
         return query;
     }
 
-    QueryResponse performQuery(SolrQuery query) throws SearchNotImplementedException {
+    QueryResponse performQuery(SolrQuery query) throws SearchServiceException {
         if (client == null) {
-            throw new SearchNotImplementedException();
+            throw new SearchServiceException("Search service is not implemented.", HttpStatus.SC_NOT_IMPLEMENTED);
         }
 
         QueryResponse response = null;
 
         try {
             response = client.query(getCollectionName(), query);
         } catch (SolrServerException e) {
-            log.severe(String.format(ERROR_SEARCH_DOCUMENT, query.getQuery(), e.getRootCause())
+            Throwable rootCause = e.getRootCause();
+            log.severe(String.format(ERROR_SEARCH_DOCUMENT, query.getQuery(), rootCause)
                     + TeammatesException.toStringWithStackTrace(e));
+            if (rootCause instanceof ConnectException) {
+                throw new SearchServiceException("Error connecting to search service.",
```

Comment:
We may not need separate error message/branch here. Unlike 4XX errors where we can tailor the message to tell users what to do to rectify the error, for 5XX errors and particularly for this case it doesn't help whether they know it's an error caused by connection (this branch) or something else (later branch); all they know is (1) it is the system's fault and not theirs and (2) it is/should be a temporary error.